src/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	tcollama "github.com/testcontainers/testcontainers-go/modules/ollama"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+)
+
+type TagResponse struct {
+	Models []struct {
+		Name string `json:"name"`
+	} `json:"models"`
+}
+
+type GenerateRequest struct {
+	Model  string `json:"model"`
+	Stream bool   `json:"stream"`
+	Prompt string `json:"prompt"`
+}
+
+func startContainer(image string) (string, error) {
+	ctx := context.Background()
+	ollamaContainer, err := tcollama.Run(ctx, image, testcontainers.CustomizeRequestOption(func(req *testcontainers.GenericContainerRequest) error {
+		req.Name = normalizeString(image)
+		req.Reuse = true
+		return nil
+	}))
+	if err != nil {
+		log.Fatalf("failed to start container: %s", err)
+		return "", err
+	}
+	return ollamaContainer.Endpoint(ctx, "http")
+}
+
+func normalizeString(input string) string {
+	// Define a regular expression that matches any character that is not alphanumeric, _, or -
+	re := regexp.MustCompile(`[^a-zA-Z0-9_-]`)
+
+	// Replace all matches with a dash
+	normalized := re.ReplaceAllString(input, "-")
+
+	return normalized
+}
+
+func generate(url string, prompt string) error {
+	tagsResp, err := http.Get(fmt.Sprintf("%s/api/tags", url))
+	if err != nil {
+		return err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
```

Comment:
nit: technically this isn't wrong (and technically probably more safe), I just have never seen people doing it like this.

You probably should be good with
```go
defer func() {
 _ = tagsResp.Close()
}()
```

Depending how your linter is configured, usually just doing
```go
defer tagsResp.Close()
```
should do it (eg check "go by example", the mostly do like this)